eclipse-che/che-server
# Add an SSH Factory parameter resolver for unsupported SCM providers

### What does this PR do?
* Return the SSH factory resolver in order to handle SSH urls from unsupported SCM providers.
* Add a `priority` value for all factory resolvers to be able to control resolver's priority.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22488

### How to test this PR?

1. Deploy che from the pr image: `quay.io/eclipse/che-server:pr-567`
2. [Configure SSH keys](https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations) for an unsupported SCM provider e.g. gitea.com.
3. Start a workspace from SSH url of the repository.

See: workspace starts from the default devfile.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
